dynamodb-stream-reader/src/main/java/no/unit/nva/dynamodb/DynamoDBStreamHandler.java

```diff
@@ -0,0 +1,121 @@
+import elasticsearch.ElasticSearchRestClient;
+import nva.commons.utils.Environment;
+import nva.commons.utils.JacocoGenerated;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.function.UnaryOperator;
+
+public class DynamoDBStreamHandler implements RequestHandler<DynamodbEvent, String> {
+
+    private static final Logger logger = LoggerFactory.getLogger(DynamoDBStreamHandler.class);
+
+    public static final String IDENTIFIER = "identifier";
+    public static final String DATE_YEAR = "entityDescription.date.year";
+    public static final String DESCRIPTION_MAIN_TITLE = "entityDescription.mainTitle";
+    public static final String CONTRIBUTORS_IDENTITY_NAME = "entityDescription.contributors.identity.name";
+    public static final String PUBLICATION_TYPE = "publicationInstance.type";
+    public static final String YEAR = "year";
+    public static final String TITLE = "title";
+    public static final String NAME = "author";
+    public static final String TYPE = "publicationType";
+    private final ElasticSearchRestClient elasticSearchClient;
+
+    /**
+     * Default constructor for DynamoDBStreamHandler.
+     */
+    @JacocoGenerated
+    public DynamoDBStreamHandler() {
+        this(new ElasticSearchRestClient(HttpClient.newHttpClient(), new Environment()));
+    }
+
+    /**
+     * Constructor for DynamoDBStreamHandler for testing.
+     * @param elasticSearchRestClient elasticSearchRestClient to be injected for testing
+     */
+    @JacocoGenerated
+    public DynamoDBStreamHandler(ElasticSearchRestClient elasticSearchRestClient) {
+        this.elasticSearchClient = elasticSearchRestClient;
+    }
+
+
+
+    @Override
+    public String handleRequest(DynamodbEvent event, Context context) {
+        for (DynamodbEvent.DynamodbStreamRecord streamRecord : event.getRecords()) {
+            switch (streamRecord.getEventName()) {
+                case "INSERT":
+                case "MODIFY":
+                    try {
+                        upsertSearchIndex(streamRecord);
+                    } catch (InterruptedException | URISyntaxException | IOException e) {
+                        logger.error("Error Upserting index", e);
+                    }
+                    break;
+                case "REMOVE":
+                    try {
+                        removeFromSearchIndex(streamRecord);
+                    } catch (InterruptedException | URISyntaxException | IOException e) {
+                        logger.error("Error deleting from index", e);
+                    }
+                    break;
+                default:
+                    throw new RuntimeException("Not a known operation");
+            }
+        }
+        return "Handled " + event.getRecords().size() + " records";
```

Comment:
Use a constant and a String.format() call here. String concatenation with + symbol has a bad performance when concatenating more than two Strings.

Reply:
Message changed to constant